test/compile_data/compile_data.rs

```diff
@@ -0,0 +1,9 @@
+/// Data loaded from compile data
+pub const COMPILE_DATA: &str = include_str!("compile_data.txt");
+
+/// A test that is expected to be compiled from a target that does not
+/// directly populate the `compile_data` attribute
+#[test]
```

Comment:
I didn't think it was just a pattern. In most cases, you'll have test dependencies and extra functions/structs/etc used for tests that you don't want in your binary otherwise. My understanding is the solution is to have a test-only module that contains all this so it can easily be removed. Since there's only one function here, I didn't think it was necessary since I don't think this would compile into a non-test target. I could be wrong but also have no problem adding the module